### PR TITLE
local-gadget: Add seccomp-profile advise gadget

### PR DESCRIPTION
**Pull request by blanquicet**
Fixes https://github.com/inspektor-gadget/inspektor-gadget/issues/866

### CODE REVIEW

File: pkg/gadget-collection/gadgets/advise/seccomp/gadget.go [hunk 1]
```diff
@@ -532,7 +532,7 @@ func (t *Trace) Generate(trace *gadgetv1alpha1.Trace) {
 
 	switch trace.Spec.OutputMode {
 	case gadgetv1alpha1.TraceOutputModeStatus:
-		policy := syscallArrToLinuxSeccomp(b)
+		policy := seccomptracer.SyscallArrToLinuxSeccomp(b)
```
Comment (mauriciovasquezbernal) [inline pkg/gadget-collection/gadgets/advise/seccomp/gadget.go:535]: I was thinking that the tracer should return the syscalls names directly. What do you think?

File: cmd/local-gadget/advise/seccomp-profile.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		// EventTypeAddContainer event.
+		if n == 1 {
+			mntns = containers[0].Mntns
+		}
+
+		stop := make(chan os.Signal, 1)
+		signal.Notify(stop, syscall.SIGINT, syscall.SIGTERM)
+
+		// The profile can be generated upon container termination or when user
+		// requests it through a signal.
```
Comment (mauriciovasquezbernal) [inline cmd/local-gadget/advise/seccomp-profile.go:102]: I have some doubts about the UX. What's the advantage of generating the policy when the container terminates compared to only generating it when the users press "Ctrl + C"?

**Review by mauriciovasquezbernal — COMMENTED** [review]
One comment in the UX before reviewing more.

File: cmd/local-gadget/advise/seccomp-profile.go (hunk 2 above)
Comment (blanquicet) [reply cmd/local-gadget/advise/seccomp-profile.go:102]: IMHO, when the pod terminates, and the gadget generates the policy, I believe that "Ctrl + C" is unnecessary. In addition, it doesn't make sense to me that it continues running after having already generated the policy (i.e. doing nothing). Don't you think?

File: pkg/gadget-collection/gadgets/advise/seccomp/gadget.go (hunk 1 above)
Comment (blanquicet) [reply pkg/gadget-collection/gadgets/advise/seccomp/gadget.go]: Yep, that makes sense. Let me try.

File: pkg/gadgets/advise/seccomp/tracer/tracer.go [hunk 3]
```diff
@@ -78,24 +80,41 @@ func NewTracer() (*Tracer, error) {
...
+	sort.Strings(names)
+	return names
+}
+
+func (t *Tracer) Peek(mntns uint64) []string {
 	b, err := t.seccompMap.LookupBytes(mntns)
 	if err != nil {
 		log.Errorf("Error while looking up the seccomp map: %s", err)
-		return make([]byte, C.SYSCALLS_COUNT)
+		return make([]string, 0)
```
Comment (mauriciovasquezbernal) [inline pkg/gadgets/advise/seccomp/tracer/tracer.go:104]: I think this should return the error here so the caller can handle it. (I know it's not related to this PR but I think we can take the opportunity to fix it). Same in other places.

File: cmd/local-gadget/advise/seccomp-profile.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	}
+
+	cmd := commonadvise.NewSeccompProfileCmd(runCmd)
+
+	utils.AddCommonFlags(cmd, &commonFlags)
+
+	return cmd
+}
+
+func generateProfile(tracer *seccompAdviseTracer.Tracer, mntns uint64) error {
```
Comment (mauriciovasquezbernal) [inline cmd/local-gadget/advise/seccomp-profile.go:126]: nit: `printProfile`?

File: cmd/local-gadget/advise/seccomp-profile.go (hunk 2 above)
Comment (mauriciovasquezbernal) [reply cmd/local-gadget/advise/seccomp-profile.go:102]: Right, makes sense to me then. I was thinking about cases where one wants to run multiple times a container with the same name and create a profile that includes syscalls from all runs but seems weird and too specific.

File: cmd/local-gadget/advise/seccomp-profile.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+			select {
+			case <-stop:
+				break WaitForEvent
+			case err := <-containerNotifier:
+				if err != nil {
+					return err
+				}
+				break WaitForEvent
+			}
+		}
```
Comment (mauriciovasquezbernal) [inline cmd/local-gadget/advise/seccomp-profile.go:114]: I suppose something like works the same?

```suggestion
		select {
		case <-stop:
		case err := <-containerNotifier:
			if err != nil {
				return err
			}
		}
```

**Review by mauriciovasquezbernal — COMMENTED** [review]
Some small comments around. 

I see you removed the interactive mode and the tests related, do you plan to add tests in this PR?

File: cmd/local-gadget/advise/seccomp-profile.go (hunk 2 above)
Comment (blanquicet) [reply cmd/local-gadget/advise/seccomp-profile.go:102]: Mmmmm yep, I think that's a weird use case. I would expect users will trigger all the syscalls in a single run.

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM. Only missing comment is about tests. I'm ok with adding them in a follow up PR.

**Comment by blanquicet** [comment]
> Only missing comment is about tests. I'm ok with adding them in a follow up PR.

Oops. Sorry, I missed it. Yeah, I plan to add that test back once https://github.com/inspektor-gadget/inspektor-gadget/pull/1162 is merged.

**Pull request merged**